packages/completer/src/model.ts

```diff
@@ -459,6 +458,52 @@ export class CompleterModel implements Completer.IModel {
     }));
   }
 
+  /**
+   * Lazy load missing data of item at `activeIndex`.
+   * @param {number} activeIndex - index of item
+   * @return Return `undefined` if the completion item with `activeIndex` index can not be found.
+   * Return a promise of `null` if another `resolveItem` is called. Otherwise return the
+   * promise of resolved completion item.
+   */
+  resolveItem(
+    activeIndex: number
+  ): Promise<CompletionHandler.ICompletionItem | null> | undefined {
+    const current = ++this._resolvingItem;
+    let resolvedItem: Promise<CompletionHandler.ICompletionItem>;
+    if (!this.completionItems) {
+      return undefined;
+    }
+
+    let completionItems = this.completionItems();
+    if (!completionItems || !completionItems[activeIndex]) {
+      return undefined;
+    }
+    let completionItem = completionItems[activeIndex];
+    if (completionItem.resolve) {
+      let patch: Completer.IPatch | undefined;
+      if (completionItem.insertText) {
+        patch = this.createPatch(completionItem.insertText);
+      }
+      resolvedItem = completionItem.resolve(patch);
+    } else {
+      resolvedItem = Promise.resolve(completionItem);
+    }
+    return resolvedItem
+      .then(activeItem => {
+        Object.assign(completionItem, activeItem);
```

Comment:
This is flagged by the code security scan. Would you mind changing the statement? Probably by something like `Object.keys(activeItem).forEach(key => { completionItem[key] = activeItem[key] });`

Reply:
I tried to use `Object.key` loop before but it becomes quite ugly to make TS compile 

```typescript
(Object.keys(activeItem) as Array<
  keyof CompletionHandler.ICompletionItem
>).forEach(
  <Key extends keyof CompletionHandler.ICompletionItem>(key: Key) => {
    completionItem[key] = activeItem[key];
  }
);
```